Source/ACE.Server/WorldObjects/Player_Use.cs

```diff
@@ -97,24 +69,26 @@ public void HandleActionUseWithTarget(uint sourceObjectGuid, uint targetObjectGu
                 return;
             }
 
-            if (sourceItem.WeenieType == WeenieType.Healer)
-            {
-                if (targetItem is Player player)
-                    ((Healer)sourceItem).HandleActionUseOnTarget(this, player);
-                else
-                    SendUseDoneEvent(WeenieError.YouCantHealThat);
-            }
-            else if (sourceItem.WeenieType == WeenieType.Key)
-            {
-                ((Key)sourceItem).HandleActionUseOnTarget(this, targetItem);
-            }
-            else if (sourceItem.WeenieType == WeenieType.Lockpick)
-            {
-                ((Lockpick)sourceItem).HandleActionUseOnTarget(this, targetItem);
-            }
-            else
+            switch (sourceItem.WeenieType)
```

Comment:
this entire switch, and all of these cases, are unneeded

HandleActionUseOnTarget() should be a virtual method, with the base in WorldObject calling RecipeManager

Reply:
True, but outside the scope of making sure that Healing works, again.  As you mentioned, this whole code limb, if you will, should probably have the SendUseDone() pulled back from the individual branches.